### DISCUSSION TITLE
Pixelation / Censoring

### DISCUSSION THREAD

**Discussion by hendursaga** [libvips/libvips#3042] [Q&A]
Is there a way I could input the coordinates for, say, a rectangle, and pixelate / censor them?

The closest thing I could find to pixelization is [https://github.com/daschr/pixelate](https://github.com/daschr/pixelate) but it's not exactly what I need, although it might have some clues as to how to proceed.

**Comment by jcupitt** [comment]
Hi @hendursaga,

You can use SVG to make a mask, then use that to flip between a pixelated and non-pixelated version of the image. See for example: https://github.com/libvips/libvips/discussions/2271

```python
#!/usr/bin/python3

import sys
import pyvips

image = pyvips.Image.new_from_file(sys.argv[1])

# make a pixelated version of the image
pixelated = image.resize(0.05).resize(20, kernel="nearest")

# make a mask
cx = image.width / 2
cy = image.height / 2
r = min(cx, cy)
mask = pyvips.Image.svgload_buffer(bytes(f"""
<svg viewBox="0 0 {image.width} {image.height}">
    <circle cx="{cx}" cy="{cy}" r="{r}" fill="#000" />
</svg>
""", "utf8"))

# svgload always makes RGBA ... we want the alpha
alpha = mask[3]

# use the mask to chose pixels from the pixelated or non-pielated versions
image = alpha.ifthenelse(pixelated, image)

image.write_to_file(sys.argv[2])
```
